test/support/test_schema.ex

```diff
@@ -0,0 +1,76 @@
+defmodule ChannelSpec.SocketTest.LotsOfRefsSchema do
```

Comment:
I'd suggest creating modules in-place in the tests instead of helper modules, like here: https://github.com/felt/channel_spec/blob/main/test/channel_spec/socket_test.exs#L51-L69

Reply:
👋 ! Any specific reason? I broke it out since it was 75+ lines of code and that seemed like a lot to put in a test

Reply:
I find it easier to have the while setup in the test itself in macro heavy code to  both have the full picture and making sure unrelated code is not affecting the generated code.
It's a matter of preference, but macros don't always produce helpful stacktraces so I'd rather avoid sharing modules in tests, and at least have the certainty that I know exactly which test is causing issues

Reply:
That's true. However, the nested nature of the setup for this test doesn't allow us to make nested modules with refs. Example:
```elixir
test "foo", %{mod: mod} do 
    # I can define the top level name
    defmodule :"#{mod}.LotsOfRefsSchema" do
      # And I can create modules in this namespace
      defmodule Base do
        def schema() do
          %{
            type: :object,
            properties: %{
              # But I can't reference it here — #{mod} is not available
              foo: %{"$ref": :"#{mod}.LotsOfRefsSchema.Foo"},
            },
            additionalProperties: false
          }
        end
      end

      defmodule Foo do
        def schema() do
          %{type: :object}
        end
      end
    end
end
```

Reply:
```
error: undefined variable "mod"
  test/channel_spec/socket_test.exs:613: Test776.LotsOfRefsSchema.Base.schema/0
     ** (CompileError) test/channel_spec/socket_test.exs: cannot compile module Test776.LotsOfRefsSchema.Base (errors have been logged)
```

Reply:
Hmm that's a good point. You could move the `defmodule Foo` inside  `defmodule Base` and use `__MODULE__.Foo` instead, or not nest them(you'd repeat `.LotsofRefsSchema` quite a bit) but it's a good point still.

I'm fine with either approach seeing a helper module may be easier

Reply:
I just pushed a commit. I was able to get it to work with a `@mod_base Module.split(__MODULE__) |> Enum.drop(-1) |> Module.concat()`

Reply:
Oh nice trick! 💜